src/common/LoadingButton.tsx

```diff
@@ -0,0 +1,32 @@
+export interface ILoadingButtonProps {
+  buttonId: string;
+  buttonClasses: string[];
+  buttonText: string;
+  buttonOnClick: () => {};
+  buttonEnabled: boolean;
+}
+
+const LoadingButton:
+React.FC<ILoadingButtonProps> = function LoadingButton(props: ILoadingButtonProps) {
+  const {
+    buttonId, buttonClasses, buttonText, buttonOnClick, buttonEnabled,
+  } = props;
+
+  if (buttonEnabled) {
+    return (
+      <button id={buttonId} className={buttonClasses.join(' ')} type="button" onClick={buttonOnClick}>{buttonText}</button>
+    );
+  }
+
+  return (
```

Comment:
This spinner is exactly what we needed, except that it causes the button's size to change, which can cause a change in the entire layout (like on the Simulation Results page, for instance).

Is there a way to accomplish this without changing the button size?

Reply:
Resolved via absolute positioning and other CSS trickery.